releasenotes/notes/cleanup-curve-analysis-96d7ff706cae5b4e.yaml

```diff
@@ -0,0 +1,31 @@
+---
+features:
+  - |
+    :class:`BaseCurveAnalysis` class has been added as a superclass of :class:`CurveAnalysis`.
```

Comment:
I would probably put this under upgrade heading as well, since it's not something a typical user will use directly and it is more related to how the existing `CurveAnalysis` class should be used when subclassing.